### PR TITLE
DAOS-5925 logging: Correct usage of DF_UUID/DP_UUID.

### PR DESCRIPTION
**Pull request by ashleypittman**
There were cases where DP_UUID wasn't used, resulting in invalid
data being written to the log file which was causing test errors
but on inspection there were also a number of places where
DF_UUID wasn't being used correctly either so fix those.

This fixes at least one logging error which was causing test failures
and could have caused a crash, and it resolves a number of places
where incorrect data would have been logged.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/container/srv_target.c [hunk 1]
```diff
@@ -1579,8 +1579,8 @@ ds_cont_tgt_force_close(uuid_t cont_uuid)
 {
 	int rc;
 
-	D_DEBUG(DF_DSMS, DF_CONT": Force closing all handles for container "
-		DF_UUID"\n", DP_CONT(NULL, NULL), cont_uuid);
```
Comment (ashleypittman) [inline src/container/srv_target.c:1583]: This is the cause of the most recent error, we were logging a raw uuid as a string.  This code is about five months old, so I can only conclude something else has landed elsewhere where we're now taking this code-path where we weren't before.

**Review by NiuYawei — APPROVED** [review]

File: src/tests/suite/daos_degraded.c [hunk 2]
```diff
@@ -84,7 +84,8 @@ insert_lookup_enum_with_ops(test_arg_t *arg, int op_kill)
 
 	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
 	if (!rank) {
-		print_message("Using pool: %s\n", DP_UUID(arg->pool.pool_uuid));
+		print_message("Using pool: " DF_UUID "\n",
```
Comment (ashleypittman) [inline src/tests/suite/daos_degraded.c:87]: This is a slight behaviour change as DF_UUID only prints the first eight characters of the uuid

File: src/tests/suite/daos_nvme_recovery.c [hunk 3]
```diff
@@ -104,7 +104,7 @@ nvme_recov_1(void **state)
 	for (i = 0; i < ndisks; i++) {
 		if (devices[i].rank != rank)
 			continue;
-		print_message("Rank=%d UUID=%s state=%s host=%s tgts=",
+		print_message("Rank=%d UUID=" DF_UUID " state=%s host=%s tgts=",
```
Comment (ashleypittman) [inline src/tests/suite/daos_nvme_recovery.c:107]: This is also a output change

File: src/tests/suite/daos_test_common.c [hunk 4]
```diff
@@ -843,7 +843,7 @@ daos_dmg_pool_target(const char *sub_cmd, const uuid_t pool_uuid,
 	int		rc;
 
 	/* build and invoke dmg cmd */
-	dts_create_config(dmg_cmd, "dmg pool %s -i --pool=%s --rank=%d",
+	dts_create_config(dmg_cmd, "dmg pool %s -i --pool=" DF_UUIDF " --rank=%d",
```
Comment (ashleypittman) [inline src/tests/suite/daos_test_common.c:846]: This appears to be creating a command line to use (although it's doing it in an odd way), so I used DF_UUIDF here which should print the whole string, so this shouldn't cause a regression.

**Review by ashleypittman — COMMENTED** [review]
This PR had multiple rebuild failures, more than appear in master and it's not clear to be why as it should only affect the logging.  Never the less I've filed https://github.com/daos-stack/daos/pull/4154 which only attempts to address the particular issue that CI is picking up on, we possibly want to land that, have someone look at rebuild and disable/revert if necessary and then circle back to this PR.

**Review by tanabarr — APPROVED** [review]

File: src/container/srv_target.c [hunk 5]
```diff
@@ -1579,8 +1579,8 @@ ds_cont_tgt_force_close(uuid_t cont_uuid)
...
+	D_DEBUG(DF_DSMS, "Force closing all handles for container "
+		DF_UUID"\n", DP_UUID(cont_uuid));
```
Comment (ashleypittman) [inline src/container/srv_target.c:1583]: This has landed in a different PR, although I'm not going to rebase as it would take time to test again.

**Comment by ashleypittman** [comment]
It passed a 2nd time, I guess some of the test logging ones are parsed by the python test code which was causing the earlier failures.

**Review by tanabarr — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]
